Sources/DuetBot/Core/Command.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  Command.swift
+//  DuetBot
+//
+//  Created by ライアン on 11/10/18.
+//
+
+import Foundation
+import Sword
+
+// NOTE: If there are more ways to reply than just with a message, turn
+// reply into a real type and return that instead.
+typealias Reply = String
+
+struct Command {
+    
+    let pattern: String
+    let arguments: [String]
+    private let message: Message
+    
+    var sender: User? {
+        return message.author
+    }
+    
+    init?(_ message: Message) {
+        self.message = message
+        guard message.content.hasPrefix("!") else { return nil }
+        let components = message.content.dropFirst().components(separatedBy: " ")
```

Comment:
How could we do this in a way where we can support a sentence as a parameter? like !command Hello There!

Reply:
Depends on how you design it, but maybe a `Parser` type which each Routine owns. A `Parser` parses the stuff after the command before observing the command .